cpp/src/io/parquet/page_enc.cu

```diff
@@ -1317,16 +1342,25 @@ __global__ void __launch_bounds__(128, 8)
       __syncthreads();
     }
   }
+
+  uint32_t const valid_count = block_reduce(temp_storage.reduce_storage).Sum(num_valid);
+
   if (t == 0) {
+    s->page.num_nulls     = s->page.num_values - valid_count;
     uint8_t* base         = s->page.page_data + s->page.max_hdr_size;
     auto actual_data_size = static_cast<uint32_t>(s->cur - base);
     if (actual_data_size > s->page.max_data_size) {
       CUDF_UNREACHABLE("detected possible page data corruption");
     }
     s->page.max_data_size = actual_data_size;
     if (not comp_in.empty()) {
-      comp_in[blockIdx.x]  = {base, actual_data_size};
-      comp_out[blockIdx.x] = {s->page.compressed_data + s->page.max_hdr_size, 0};  // size is unused
+      // V2 does not compress rep and def level data
+      size_t const offset  = s->page.def_lvl_bytes + s->page.rep_lvl_bytes;
```

Comment:
Took me a bit to connect the dots in this block. Maybe a rename?
```suggestion
      size_t const skip_comp_size = s->page.def_lvl_bytes + s->page.rep_lvl_bytes;
```

Reply:
sounds good.  Oh, the memcpy just below...I'm thinking that should be parallelized, but keep forgetting.  Thoughs?  The level data can get pretty large (100s of bytes?).